src/Components/Coins/CoinsButton.js

```diff
@@ -1,6 +1,29 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import CoinsTab from './CoinsTab';
 import './CoinsButton.css';
+import { Button } from '@material-ui/core';
+import { MountedComponentsContext } from 'Contexts/MountedComponentsContext';
+import { componentsTitles } from 'Texts/texts';
 
 export default function CoinsButton() {
-  return <div id="coinsButton">Coins</div>;
+  const { mountedComponentsDict, mountComponents, unmountComponents } = useContext(MountedComponentsContext);
+
+  const isCoinsTabOpen = mountedComponentsDict.secondaries.CoinsTab;
+
+  const onCoinsButtonClick = () => {
+    if (isCoinsTabOpen) {
+      unmountComponents(['CoinsTab', 'CoinsPurchaseWindow']);
```

Comment:
see with eyal if this is the best practice (to unmount CoinsPurchaseWindow in any case)

Reply:
nah it's cool